### PR TITLE
Cleanup and preparation for sim extentions

### PR DESCRIPTION
**Pull request by civanch**
#### PR description:
Number of reorganisations inside Geant4 user actions and elements of code guideline in some classes. The main purpose is to make further R&D efforts easier. Tracking, Stepping and Stacking action use SimTrackManager directly and not EventAction as an intermediate object.

No change in any WF is expected.

#### PR validation:
private

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimG4Core/Application/src/RunManagerMTWorker.cc [hunk 1]
```diff
@@ -131,15 +132,15 @@ struct RunManagerMTWorker::TLSData {
 
   TLSData() {}
 
-  ~TLSData() {}
+  ~TLSData() = default;
```
Comment (perrotta) [inline SimG4Core/Application/src/RunManagerMTWorker.cc:135]: please move to RunManagerMTWorker.h

File: SimG4Core/CustomPhysics/plugins/DBremWatcher.cc [hunk 2]
```diff
@@ -1,4 +1,47 @@
...
+  double biasFactor;
+  bool foundbrem;
+  G4ThreeVector aPrimeTraj;
+  G4ThreeVector finaltraj;
+  G4ThreeVector VertexPos;
+  float f_energy;
+};
 
 #include "SimG4Core/Notification/interface/BeginOfTrack.h"
 #include "SimG4Core/Notification/interface/BeginOfEvent.h"
```
Comment (perrotta) [inline SimG4Core/CustomPhysics/plugins/DBremWatcher.cc:47]:
```suggestion
```
Duplicate headers (quite likely a few more are duplicated from what was previously in the separate header file)

Comment (civanch) [reply SimG4Core/CustomPhysics/plugins/DBremWatcher.cc]: Fixed

**Comment by civanch** [comment]
please test

File: SimG4Core/Application/src/RunManagerMTWorker.cc (hunk 1 above)
Comment (civanch) [reply SimG4Core/Application/src/RunManagerMTWorker.cc:135]: Likely no need to move to the header.

Comment (perrotta) [reply SimG4Core/Application/src/RunManagerMTWorker.cc:135]: @makortel @civanch you are correct, it was a mistake of mine during the review

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**